pkg/v2/service.go

```diff
@@ -565,8 +565,37 @@ func (s *service) Shutdown(ctx context.Context, r *taskAPI.ShutdownRequest) (*pt
 }
 
 func (s *service) Stats(ctx context.Context, r *taskAPI.StatsRequest) (*taskAPI.StatsResponse, error) {
-	// TODO(random-liu): Use `runsc events --stats`.
-	data, err := typeurl.MarshalAny(&cgroups.Metrics{})
+	path, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+	ns, err := namespaces.NamespaceRequired(ctx)
+	if err != nil {
+		return nil, err
+	}
+	runtime, err := s.readRuntime(path)
+	if err != nil {
+		return nil, err
+	}
+	rs := proc.NewRunsc(proc.RunscRoot, path, ns, runtime, nil)
+	stats, err := rs.Stats(ctx, s.id)
+	if err != nil {
+		return nil, err
+	}
+
+	// gvisor currently (as of 2020-02-28) only returns the total memory
+	// usage and current PID value. Copy those into the containerd cgroup
+	// statistics structure so that we're returning the same type as runc.
+	data, err := typeurl.MarshalAny(&cgroups.Metrics{
```

Comment:
Can you fill up all fields in cgroups.Metrics? This way, when new data is returned from runsc, they are automatically propagated from here.

Reply:
Done!